### DISCUSSION TITLE
Langfue API accessing

### DISCUSSION THREAD

**Discussion by Harish-ilink** [langfuse/langfuse#4771] [Support] [answered]
I want to integrate Langfuse login with my project. To do this, I need to log in using Python code and retrieve the necessary cookies and API key to access various APIs, such as fetching prompts or traces. I used the endpoint "api/auth/callback/credentials" for logging in and provided a username and password with basic authentication as the authorization method. However, instead of receiving a JSON response, I got an HTML page as the response, and I am unable to access the response cookies. Could someone guide me on resolving this issue?

![image](https://github.com/user-attachments/assets/469092e1-451d-4e01-91bd-6c03603b641d)

**Comment by marcklingen** [comment, marked as answer]
The `api/auth/callback/credentials` API is intended for use by the frontend only. It is not designed for direct access and is not covered by semantic versioning, meaning it may break in minor versions.

To access the public API, you need the public and secret keys of a project, which will allow you to authenticate with the public API. For more details, visit: https://api.reference.langfuse.com

**Reply by Harish-ilink** [reply]
I have a self hosted a project How can i use these api references?

**Reply by Harish-ilink** [reply]
For eg to access apis we need api key right how can i get that in to access langfuse APIs?

**Reply by marcklingen** [reply]
You can get the API keys in the project settings when logging into the UI.

Alternatively, you can provision default credentials for a default project, more details here: https://langfuse.com/self-hosting/headless-initialization
